CriteoPublisherSdk/Sources/Network/CR_ApiHandler.m

```diff
@@ -138,26 +139,33 @@ - (void)doCdbApiCall:(CR_CacheAdUnitArray *)adUnits
                                                                 config:config
                                                             deviceInfo:deviceInfo
                                                                context:contextData];
-    CLogInfo(@"[INFO][API_] CdbPostCall.start");
-    [self.networkManager postToUrl:url
-                          postBody:body
-                   responseHandler:^(NSData *data, NSError *error) {
-                     CLogInfo(@"[INFO][API_] CdbPostCall.finished");
-                     if (error == nil) {
-                       if (completionHandler) {
-                         CR_CdbResponse *cdbResponse = [self cdbResponseWithData:data];
-                         completionHandler(cdbRequest, cdbResponse, error);
-                       }
-                     } else {
-                       CLog(@"Error on post to CDB : %@", error);
-                       if (completionHandler) {
-                         completionHandler(cdbRequest, nil, error);
-                       }
-                     }
-                     for (CR_CacheAdUnit *adUnit in adUnitChunk) {
-                       [self.bidFetchTracker clearBidFetchInProgressForAdUnit:adUnit];
-                     }
-                   }];
+    NSData *postData =
+        [self.networkManager postToUrl:url
+                              postBody:body
+                       responseHandler:^(NSData *data, NSError *error) {
+                         if (data) {
+                           CRLogInfo(@"BidResponse", @"⬇️ Received: %@",
+                                     [NSString stringWithUTF8String:[data bytes]]);
+                         }
+                         if (error == nil) {
+                           if (completionHandler) {
+                             CR_CdbResponse *cdbResponse = [self cdbResponseWithData:data];
+                             completionHandler(cdbRequest, cdbResponse, error);
+                           }
+                         } else {
+                           CRLogInfo(@"BidResponse", @"⬇️ Error: %@", error);
+                           if (completionHandler) {
+                             completionHandler(cdbRequest, nil, error);
+                           }
+                         }
+                         for (CR_CacheAdUnit *adUnit in adUnitChunk) {
+                           [self.bidFetchTracker clearBidFetchInProgressForAdUnit:adUnit];
+                         }
+                       }];
+    if (postData) {
```

Comment:
In what case can `postData` be nil?

Reply:
On nil data / json serialisation error, which both are error cases that should be handled / logged differently.